pkg/testutil/leak.go

```diff
@@ -113,6 +113,7 @@ func interestingGoroutines() (gs []string) {
 			strings.Contains(stack, "github.com/coreos/etcd/pkg/testutil.interestingGoroutines") ||
 			strings.Contains(stack, "github.com/coreos/etcd/pkg/logutil.(*MergeLogger).outputLoop") ||
 			strings.Contains(stack, "github.com/golang/glog.(*loggingT).flushDaemon") ||
+			strings.Contains(stack, "github.com/coreos/etcd/integration.TestV3ClientMetrics") ||
```

Comment:
@heyitsanthony i undertand your concern, i had initially mentioned a double check for this for net/http induced go routine as well as the current check, later i had modified this check just only to check if semaphore passes. But to no avail. As i had mentioned earlier as well i had written a custom code by passing listener to this, but on closing the listener the entire test was failing. Also i didn't find an alternative to close the go-routine i was setting up, But to no avail. Had asked for help from you as well for this but i guess you we busy with other work.As a last resort i had to dig deep and found an alternative this way. Would be grateful to you if you could help me out on closing the go routine which i am creating as i haven't been able to figure out an alternative to close this.
Thanks a lot for looking into this.